CI: macOS build broken by Homebrew LLVM 22.1.1 (apache-arrow dependency)

## Problem

macOS CI builds are failing with `cmath` errors since 2026-03-19:

```
/opt/homebrew/bin/../include/c++/v1/cmath:544:5: error: use of undeclared identifier '__promote'
/opt/homebrew/bin/../include/c++/v1/cmath:548:22: error: no template named '__promote'
/opt/homebrew/bin/../include/c++/v1/cmath:552:12: error: call to 'hypot' is ambiguous
```

The errors occur in generic files (`MockImplementation.cpp`, `ClassTest.cpp`) — not specific to any PR.

## Root Cause

`brew install apache-arrow` now pulls in **LLVM 22.1.1** as a dependency:
```
Installing apache-arrow dependency: llvm
Pouring llvm--22.1.1.arm64_sonoma.bottle.tar.gz
/opt/homebrew/Cellar/llvm/22.1.1: 9,686 files, 1.9GB
```

The CI workflow sets the compiler via:
```bash
cxx_compiler=$(brew --prefix llvm@15.4)/bin/clang++
```

This resolves to `/opt/homebrew/bin/clang++` which is now Homebrew's **LLVM 22.1.1** (not Apple Clang from Xcode 15.4). LLVM 22's libc++ headers (`/opt/homebrew/bin/../include/c++/v1/cmath`) are incompatible with the Xcode 15.4 macOS SDK headers, causing `__promote` to be undefined.

## Evidence

- **Last passing macOS CI on develop:** 2026-03-19T01:43 (run 23275760267)
- **First failing:** 2026-03-19T12:49 (after Homebrew updated LLVM)
- **Affected PRs:** #8939, #8940 (any PR that triggers a fresh macOS build today)
- **Unaffected PRs:** #8938, #8944 (ran before the LLVM update)

## Suggested Fix

Pin the macOS compiler to Apple Clang instead of Homebrew LLVM. In `.github/workflows/openms_ci_matrix_full.yml`, change the xcode compiler setting to use the Xcode toolchain explicitly:

```yaml
# Instead of:
echo "cxx_compiler=$(brew --prefix llvm@15.4)/bin/clang++" >> $GITHUB_OUTPUT

# Use Apple Clang from Xcode:
echo "cxx_compiler=/usr/bin/clang++" >> $GITHUB_OUTPUT
```

Alternatively, prevent Homebrew from installing its own LLVM by using `apache-arrow` from a pre-built source that doesn't depend on LLVM, or by adding `brew pin llvm` after the Xcode-selected version is set.

## Environment

- Runner: `macos-14` (ARM64)
- Xcode: 15.4
- Homebrew LLVM: 22.1.1 (installed as apache-arrow dependency)
- Failing compile line: `ccache /opt/homebrew/bin/clang++ ... -std=gnu++20`

## Investigation results

After pulling the actual CI logs (run 23283027293), the root cause described in this issue is **incorrect**.

### What the logs show

The compiler **is** Apple Clang (`/usr/bin/clang++`), not Homebrew LLVM:
```
ccache /usr/bin/clang++ ... -std=gnu++20 ...
```

There is exactly **one build error**, and it's a code bug, not a toolchain issue:
```
src/openms/source/MATH/STATISTICS/KernelDensityEstimation.cpp:243:28: error: no member named '__bit_ceil' in namespace 'std'; did you mean 'bit_ceil'?
      std::size_t M = std::__bit_ceil(target);
```

`std::__bit_ceil` is a libc++ internal — Apple's libc++ doesn't expose it. The standard API is `std::bit_ceil`.

### Why it's branch-specific

| Branch | macOS Build | Explanation |
|--------|------------|-------------|
| `develop` (nightly) | **passes** (fails later at Notarize) | Has correct `std::bit_ceil` |
| `remove-qt/pr3-filesystem` | **passes** (7/7) | Doesn't modify this file |
| `cleanup/clang-tidy-performance-emplace-bugprone` | fails (3/3) | Likely clang-tidy replaced `bit_ceil` → `__bit_ceil` |
| `remove-qt/pr4-process` | fails (3/3) | Same issue |

### Conclusion

This is not a Homebrew LLVM / compiler PATH issue. The `xcode-select` + bare `clang++` setup works correctly — it resolves to `/usr/bin/clang++`. The failing branches have a code-level bug where `std::__bit_ceil` (non-standard internal) is used instead of `std::bit_ceil`.

--- a/.github/workflows/openms_ci_matrix_full.yml
+++ b/.github/workflows/openms_ci_matrix_full.yml
@@ -151,7 +151,7 @@ jobs:
           if [[ "${{ matrix.compiler }}" == "xcode" ]]; then
             ls -la /Applications/Xcode*
             sudo xcode-select -s '/Applications/Xcode_${{ matrix.compiler_ver }}.app/Contents/Developer'
-            echo "cxx_compiler=clang++" >> $GITHUB_OUTPUT
+            echo "cxx_compiler=$(xcrun -f clang++)" >> $GITHUB_OUTPUT
           elif [[ "${{ matrix.compiler }}" == "clang++" ]]; then
             echo "cxx_compiler=$(brew --prefix llvm@${{ matrix.compiler_ver }})/bin/clang++" >> $GITHUB_OUTPUT
           else

